Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/EditorIntrinsicsCalibration.cs

```diff
@@ -0,0 +1,180 @@
+                }
+                else
+                {
+                    CalibrationDataHelper.SaveChessboardDetectedImage(texture, i);
+                }
+            }
+
+            if (chessboardHeatmap)
+                CalibrationDataHelper.SaveImage(chessboardHeatmap, "ChessboardHeatmap");
+
+            if (chessboardCorners)
+                CalibrationDataHelper.SaveImage(chessboardCorners, "ChessboardCorners");
+        }
+
+        private void Update()
+        {
+            if (feedImage != null &&
+                feedImage.texture == null)
+                feedImage.texture = CompositorWrapper.GetDSLRFeed();
+
+            if (cornersImage)
+                cornersImage.texture = chessboardCorners;
+
+            if (heatmapImage)
+                heatmapImage.texture = chessboardHeatmap;
+
+            if (Input.GetKeyDown(KeyCode.Space))
+            {
+                var dslrTexture = CompositorWrapper.GetDSLRTexture();
+                CalibrationDataHelper.SaveChessboardImage(dslrTexture, nextChessboardImageId);
+
+                if(!ProcessChessboardImage(dslrTexture))
+                {
+                    Debug.LogWarning($"Failed to process/locate chessboard in dataset: {nextChessboardImageId}");
+                }
+                else
+                {
+                    CalibrationDataHelper.SaveChessboardDetectedImage(dslrTexture, nextChessboardImageId);
+                    CalibrationDataHelper.SaveImage(chessboardHeatmap, "ChessboardHeatmap");
+                    CalibrationDataHelper.SaveImage(chessboardCorners, "ChessboardCorners");
+                }
+
+                nextChessboardImageId++;
+            }
+
+            if (Input.GetKeyDown(KeyCode.Return))
+            {
+                Debug.Log("Starting Camera Intrinsics calculation.");
+                intrinsics = calibration.CalculateChessboardIntrinsics(chessSquareSize);
+                foreach(var intrinsic in intrinsics)
+                {
+                    Debug.Log($"Chessboard intrinsics reprojection error: {intrinsic.ToString()}");
+                    var file = CalibrationDataHelper.SaveCameraIntrinsics(intrinsic);
+                    Debug.Log($"Camera Intrinsics saved to file: {file}");
+                }
+            }
+        }
+
+        private bool ProcessChessboardImage(Texture2D texture)
+        {
+            if (texture == null ||
+                texture.format != TextureFormat.RGB24)
+            {
+                return false;
+            }
+
+            int imageWidth = texture.width;
+            int imageHeight = texture.height;
+
+            if (chessboardCorners == null)
+            {
+                chessboardCorners = new Texture2D(imageWidth, imageHeight, TextureFormat.RGB24, false);
+                for (int y = 0; y < chessboardCorners.height; y++)
+                {
+                    for (int x = 0; x < chessboardCorners.width; x++)
+                    {
```

Comment:
This is a really weird question but maybe mentally I'm so used to going where I iterate over x first, then y, then z (or i -> j -> k) - is this order primarily because you get some sort of perf benefits of memory localility, or is it also equally arbitrary like what I usually do?

Reply:
I think that these values are stored in rows compared to columns, so my assumption would be that accessing subsequent x values would have better performance than different y values. That going said, this was just an arbitrary choice.